### PR TITLE
Enhancement of `configure.py`

### PR DESCRIPTION
**Pull request by ChunYen-Chen**
According to #216, I changed the following things:
- Replace `== None` with `is None`.
- Add the shebang, and modify the file mode to be executable.
- Use `os.path.join()` to combine path names.
- Disable the autofill feature (`allow_abbrev`) across all versions of Python.

Additionally, I changed the following to make the code easier to maintain:
- Optimize `print_usage`.
- Use `logging` to control the output message.
- Use the `path_links` in `warning()` to reduce the redundant code.

Moreover, I added the following features:
- Record the executed command at the top of  `Makefile`.
- Recore the output message at `Makefile.out`.

### CODE REVIEW

File: src/configure.py [hunk 1]
```diff
@@ -720,107 +741,84 @@ def validation( paths, depends, constraints, **kwargs ):
...
-        color_print("ERROR: <--patch_size> should be an even number greater than or equal to 8. Current: %d"%kwargs["patch_size"], BCOLOR.FAIL)
+        LOGGER.error("<--patch_size> should be an even number greater than or equal to 8. Current: %d"%kwargs["patch_size"])
         success = False
 
     if kwargs["overlap_mpi"]:
-        color_print("ERROR: <--overlap_mpi> is not supported yet.", BCOLOR.FAIL)
+        LOGGER.error("<--overlap_mpi> is not supported yet.")
         success = False
 
     if not success: raise BaseException(BCOLOR.FAIL+"The above validation failed."+BCOLOR.ENDC)
```
Comment (hyschive) [inline src/configure.py:800]: `BCOLOR` is undefined.

Comment (ChunYen-Chen) [reply src/configure.py]: Fixed

File: src/configure.py [hunk 2]
```diff
@@ -21,18 +23,28 @@
 NONE_STR   = "OFF"
 PYTHON_VER = [sys.version_info.major, sys.version_info.minor]
```
Comment (vivi235711) [inline src/configure.py:24]: Unused PYTHON_VER?

File: src/configure.py [hunk 3]
```diff
@@ -732,107 +753,84 @@ def validation( paths, depends, constraints, **kwargs ):
...
-        color_print("ERROR: <--model=PAR_ONLY> is not supported yet.", BCOLOR.FAIL)
+        LOGGER.error("<--model=PAR_ONLY> is not supported yet.")
         success = False
     else:
-        color_print("ERROR: Unrecognized model: <%s>. Please add to the model choices."%kwargs["model"], BCOLOR.FAIL)
+        LOGGER.error("Unrecognized model: %s. Please add to the model choices."%kwargs["model"])
         success = False
 
-    # A.3 Particle
+#   A.3 Particle
```
Comment (vivi235711) [inline src/configure.py:783]: Where is A.2?

**Review by vivi235711 — APPROVED** [review]

File: src/configure.py (hunk 3 above)
Comment (ChunYen-Chen) [reply src/configure.py]: Fixed

File: src/configure.py (hunk 2 above)
Comment (ChunYen-Chen) [reply src/configure.py]: Removed

**Comment by ChunYen-Chen** [comment]
@vivi235711 Thanks for the review.

**Review by hyschive — APPROVED** [review]

**Pull request merged**